docs/setup_google.md

```diff
@@ -264,49 +294,154 @@ In this part, you will learn how to setup Google login in Android
    - Copy the client ID, you'll use this as the `webClientId` in your JS/TS code
      ![](./assets/google_cons_copy_web_client_id.png)
 
-3. Modify the `AppDelegate`
-   Although not strictly required, it's recommended by Google.
+3. Now, you need to modify your `MainActivity`
    
+   - Please open your app in Android Studio. You can run `cap open android`
    
-   - Open the AppDelegate
-     ![](./assets/xcode_app_deleg.png)
+   - Please find `MainActivity.java`
+     
+     - Open the `app` folder
+       ![](./assets/studio_app_folder.png)
+     
+     - Find `java`
+       ![](./assets/studio_app_java.png)
+     
+     - Find your `MainActivity.java` and click on it**![](./assets/studio_app_java_activity_main.png)
+- Now, you have to modify `MainActivity.java`. Please add the following code
+  
+  ```java
+  @Override
+  public void onActivityResult(int requestCode, int resultCode, Intent data) {
+    super.onActivityResult(requestCode, resultCode, data);
+  
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && requestCode >= GoogleProvider.REQUEST_AUTHORIZE_GOOGLE_MIN && requestCode < GoogleProvider.REQUEST_AUTHORIZE_GOOGLE_MAX) {
+        PluginHandle pluginHandle = getBridge().getPlugin("SocialLogin");
+        if (pluginHandle == null) {
+            Log.i("Google Activity Result", "SocialLogin login handle is null");
+            return;
+        }
+        Plugin plugin = pluginHandle.getInstance();
+        if (!(plugin instanceof SocialLoginPlugin)) {
+            Log.i("Google Activity Result", "SocialLogin plugin instance is not SocialLoginPlugin");
+            return;
+        }
+        ((SocialLoginPlugin) plugin).handleGoogleLoginIntent(requestCode, data);
```

Comment:
can you explain why this is required? in what scenario?

Reply:
Both offline access and additional scopes are based on authorization. Authorization requires this change